stackslib/src/chainstate/nakamoto/test_signers.rs

```diff
@@ -124,18 +134,119 @@ impl Default for TestSigners {
+            }
+        };
+        let mut sig_aggregator = wsts::v2::Aggregator::new(num_keys, threshold);
+        sig_aggregator
+            .init(&poly_commitments)
+            .expect("aggregator init failed");
+        let aggregate_public_key = sig_aggregator.poly[0];
+        Self {
+            signer_parties,
+            aggregate_public_key,
+            poly_commitments,
+            num_keys,
+            threshold,
+            party_key_ids,
+            cycle: 0,
+            signer_keys,
+        }
+    }
+
+    /// Sign a Nakamoto block using [`Self::signer_keys`].
+    ///
+    /// N.B. If any of [`Self::signer_keys`] are not in the reward set, the resulting
+    /// signatures will be invalid. Use [`Self::sign_block_with_reward_set()`] to ensure
+    /// that any signer keys not in the reward set are not included.
     pub fn sign_nakamoto_block(&mut self, block: &mut NakamotoBlock, cycle: u64) {
         // Update the aggregate public key if the cycle has changed
         if self.cycle != cycle {
             self.generate_aggregate_key(cycle);
         }
 
-        let mut rng = rand_core::OsRng;
+        // TODO: epoch gate for aggregated signatures
+        // let signer_signature = self.sign_block_with_aggregate_key(&block);
+
+        let signer_signature = self.generate_block_signatures(&block);
+
+        test_debug!(
+            "Signed Nakamoto block {} with {} signatures (rc {})",
+            block.block_id(),
+            signer_signature.len(),
+            cycle
+        );
+        block.header.signer_signature = signer_signature;
+    }
+
+    /// Sign a NakamotoBlock and maintain the order and membership
+    /// of the reward set signers in the resulting signatures.
+    ///
+    /// If any of [`Self::signer_keys`] are not in the reward set, their signatures
+    /// will not be included.
+    pub fn sign_block_with_reward_set(&self, block: &mut NakamotoBlock, reward_set: &RewardSet) {
+        let signatures = self.generate_ordered_signatures(block, reward_set);
+        block.header.signer_signature = signatures;
+    }
+
+    /// Sign a Nakamoto block and generate a vec of signatures. The signatures will
+    /// be ordered by the signer's public keys, but will not be checked against the
+    /// reward set.
+    fn generate_block_signatures(&self, block: &NakamotoBlock) -> Vec<MessageSignature> {
+        let msg = block.header.signer_signature_hash().0;
+        let mut keys = self.signer_keys.clone();
+        keys.sort_by(|a, b| {
+            let a = Secp256k1PublicKey::from_private(a).to_bytes_compressed();
+            let b = Secp256k1PublicKey::from_private(b).to_bytes_compressed();
+            a.cmp(&b)
+        });
+        keys.iter().map(|key| key.sign(&msg).unwrap()).collect()
+    }
+
+    /// Sign a Nakamoto block using the aggregate key.
+    /// NB: this function is current unused.
+    fn sign_block_with_aggregate_key(&mut self, block: &NakamotoBlock) -> ThresholdSignature {
```

Comment:
As this is unused and a private function, won't this make clippy complain? Might be good to annotate the function with `#[allow(dead_code]`